server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1710,7 +1710,7 @@ private void cacheRepositoryData(BytesReference updated, long generation) {
         if (cacheRepositoryData && bestEffortConsistency == false) {
             final BytesReference serialized;
             try {
-                serialized = CompressorFactory.COMPRESSOR.compress(updated);
+                serialized = CompressorFactory.DEFLATE_COMPRESSOR.compress(updated);
```

Comment:
Hm ... did we missed that? Shouldn't we use `compressor` here? : compressor.compress(updated);

```
                serialized = compressor.compress(updated);

```

Reply:
fixed